File: MyApplication/app/src/main/java/com/example/alexander/myapplication/MainActivity.java
package com.example.alexander.myapplication;

import android.content.Context;
import android.graphics.Bitmap;
import android.os.Bundle;
import android.os.Environment;
import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.ListView;

import java.io.File;

public class MainActivity extends AppCompatActivity {

    ListView mLvImages;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mLvImages = (ListView) findViewById(R.id.lvImages);


        File dir = new File(Environment.getExternalStorageDirectory(), "Download/L0161");
        File[] filesArray = dir.listFiles();

        if (filesArray != null) {
            ImageAdapter adapter = new ImageAdapter(this, filesArray);
            mLvImages.setAdapter(adapter);
        }

    }


    static class ImageAdapter extends ArrayAdapter<File> {

        LayoutInflater mInflater;
        int mSize;

        public ImageAdapter(Context context, File[] objects) {
            super(context, R.layout.list_item, objects);
            mInflater = LayoutInflater.from(context);
            mSize = context.getResources().getDimensionPixelSize(R.dimen.image_size);
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {
            View view = convertView;
            if (view == null) {
                view = mInflater.inflate(R.layout.list_item, parent, false);
            }
            ImageView imageView = (ImageView) view.findViewById(R.id.imageView);
            Bitmap bitmap = getBitmap(position);
            imageView.setImageBitmap(bitmap);
            return view;
        }

        private Bitmap getBitmap(int position) {
            String filePath = getItem(position).getAbsolutePath();
            return Utils.decodeSampledBitmapFromResource(filePath, mSize, mSize);
        }


    }
}

File: Decorator/app/src/main/java/com/example/alexander/decorator/MainActivity.kt
package com.example.alexander.decorator


import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.ExpandableListView
import android.widget.SimpleExpandableListAdapter
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private var adapter: SimpleExpandableListAdapter? = null
    private var beverageName: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        buttonBegin.setOnClickListener {
            setContentView(R.layout.expandeblelist)
            var ah = AdapterHelper(this)
            adapter = ah.getAdapter()

            var elvMain = findViewById(R.id.elvMain) as ExpandableListView
            elvMain.setAdapter(adapter)
            elvMain.setOnChildClickListener(object : ExpandableListView.OnChildClickListener {
                override fun onChildClick(parent: ExpandableListView, v: View, groupPosition: Int, childPosition: Int, id: Long): Boolean {
                    beverageName = ah.getGroupChildText(groupPosition,childPosition)
                    selectPortion()
                    return false
               }
            })
        }
    }

    fun selectPortion()
    {
        when(beverageName){//тут будет помещаться тип
            "Латте", "Каппучино", "Кофе" -> {
                var intent = Intent(this, PortionTwoTypeActivity::class.java)
                intent.putExtra("beverageName", beverageName);
                startActivity(intent)
                super.finish();
            }
            "Чай" -> {
                var intent = Intent(this, ColorTeaTypeActivity::class.java)
                intent.putExtra("beverageName", beverageName);
                startActivity(intent)
                super.finish();
            }
            "Молочный коктейль" -> {
                var intent = Intent(this, MilkshakePortionTypeActivity::class.java)
                intent.putExtra("beverageName", beverageName);
                startActivity(intent)
                super.finish();
            }
            else -> { null }
        }
    }

}


